src/metal/window-proxy.ts

```diff
@@ -26,97 +24,74 @@ interface WindowProxy {
   getWidth: Function;
   rAF: Function;
   meta: MetaInterface;
-  onWindowIsDirtyListeners: OnWindowIsDirtyInterface[];
-  __destroy__: Function;
+  version: number;
+  lastVersion: number;
+  updateMeta: Function;
+  isDirty: boolean;
```

Comment:
why have an isDirty + version + lastVersion?

Reply:
isDirty is doing the validation check from version to lastVersion.

Reply:
For this PR I am not going to address this. However we will table this and include in Spaniel 4.0.0:

- Cached version state on the Window Proxy will be moved to have its own interface with a version.
- One flat object with the prefix `last` that pertain to the `lastVersion`.